server/bundler/loader.js

```diff
@@ -32,23 +30,44 @@ function getSectionsModule( sections ) {
 		'		return ' + JSON.stringify( sections ) + ';',
 		'	},',
 		'	load: function() {',
-		'		' + sectionLoaders,
+		'		' + loadFunctionBody,
+		'	},',
+		'	preload: function( section ) {',
+		'		' + preloadFunctionBody,
 		'	}',
 		'};'
 	].join( '\n' );
 }
 
-function getRequires( sections ) {
-	var content = '';
+function getPreloadSwitch( sections ) {
+	return [
+		'switch ( section ) {',
+			cases( sections ),
+		'}'
+	].join( '\n' );
 
-	sections.forEach( function( section ) {
-		content += requireTemplate( section.module );
-	} );
+	function cases( sections ) {
+		return sections.map( function( section ) {
+			return ensureCaseTemplate( section.name );
+		} ).join( '' );
+	}
```

Comment:
Personally I prefer the shorthand syntax in cases like these where no `this` is in scope. Maybe this file is a no-ES6 file - not sure.

```js
const cases = sections => sections.map( section => ensureCaseTemplate( section.name ) ).join( '' );
```